dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java

```diff
@@ -192,123 +191,128 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
+			        if ( foundContentlets != null && !foundContentlets.isEmpty() ) {
+			            //Prefer the contentlet with the session language
+			            content = foundContentlets.get(0);
+			            if ( content.getLanguageId() != lang && foundContentlets.size() == 2 ) {
+			                content = foundContentlets.get(1);
+			            }
+			        } else {
+			            Logger.error(this, "Content with Identifier [" + assetIdentifier + "] not found.");
+			            resp.sendError(404);
+			            return;
+			        }
+
+			    }
+			    else {
+			        /*
 						If the property DEFAULT_FILE_TO_DEFAULT_LANGUAGE is false OR the language in request/session
 						is equals to the default language, continue with the default behavior.
-						 */
-						content = contentAPI.findContentletByIdentifier(assetIdentifier, live, lang, user, respectFrontendRoles);
-					}
-					assetInode = content.getInode();
-				}
+			         */
+			        content = contentAPI.findContentletByIdentifier(assetIdentifier, live, lang, user, respectFrontendRoles);
+			    }
+			    assetInode = content.getInode();
+			}
 
-                // If the user is NOT logged in the backend then we cannot show content that is NOT live.
-                // Temporal files should be allowed any time
-                if(!isTempBinaryImage && !WebAPILocator.getUserWebAPI().isLoggedToBackend(req)) {
-                    if (!APILocator.getVersionableAPI().hasLiveVersion(content) && respectFrontendRoles) {
-                        Logger.debug(this, "Content " + fieldVarName + " is not publish, with inode: "
-                                + content.getInode());
-                        resp.sendError(404);
-                        return;
-                    }
-                }
+			// If the user is NOT logged in the backend then we cannot show content that is NOT live.
+			// Temporal files should be allowed any time
+			if(!isTempBinaryImage && !WebAPILocator.getUserWebAPI().isLoggedToBackend(req)) {
+			    if (!APILocator.getVersionableAPI().hasLiveVersion(content) && respectFrontendRoles) {
+			        Logger.debug(this, "Content " + fieldVarName + " is not publish, with inode: "
+			                + content.getInode());
+			        resp.sendError(404);
+			        return;
+			    }
+			}
 
-				//Find the contentlet content type
-				ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find((content.getContentTypeId()));
-				//And the file asset field
-				com.dotcms.contenttype.model.field.Field field;
+			//Find the contentlet content type
+			ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find((content.getContentTypeId()));
+			
+			if(type.baseType() != BaseContentType.FILEASSET){
+                Logger.debug(this,"Content is not a File-Type Content Type.");
+                resp.sendError(404);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 404 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)